src/shazamkit.cs

```diff
@@ -0,0 +1,250 @@
+	{
+		[Export ("frequencySkew")]
+		float FrequencySkew { get; }
+
+		[Export ("matchOffset")]
+		double MatchOffset { get; }
+
+		[Export ("predictedCurrentMatchOffset")]
+		double PredictedCurrentMatchOffset { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHMediaItem : NSSecureCoding, NSCopying
+	{
+		[NullAllowed, Export ("shazamID")]
+		string ShazamId { get; }
+
+		[NullAllowed, Export ("title")]
+		string Title { get; }
+
+		[NullAllowed, Export ("subtitle")]
+		string Subtitle { get; }
+
+		[NullAllowed, Export ("artist")]
+		string Artist { get; }
+
+		[Export ("genres", ArgumentSemantic.Strong)]
+		string[] Genres { get; }
+
+		[NullAllowed, Export ("appleMusicID")]
+		string AppleMusicId { get; }
+
+		[NullAllowed, Export ("appleMusicURL", ArgumentSemantic.Strong)]
+		NSUrl AppleMusicUrl { get; }
+
+		[NullAllowed, Export ("webURL", ArgumentSemantic.Strong)]
+		NSUrl WebUrl { get; }
+
+		[NullAllowed, Export ("artworkURL", ArgumentSemantic.Strong)]
+		NSUrl ArtworkUrl { get; }
+
+		[NullAllowed, Export ("videoURL", ArgumentSemantic.Strong)]
+		NSUrl VideoUrl { get; }
+
+		[Export ("explicitContent")]
+		bool ExplicitContent { get; }
+
+		[NullAllowed, Export ("isrc")]
+		string Isrc { get; }
+
+		[Export ("mediaItemWithProperties:")]
+		SHMediaItem MediaItem (NSDictionary<NSString, NSObject> properties);
+
+		[Async]
+		[Export ("fetchMediaItemWithShazamID:completionHandler:")]
+		void FetchMediaItem (string shazamID, Action<SHMediaItem, NSError> completionHandler);
+
+		[Export ("valueForProperty:")]
+		NSObject Value (string property);
+
+		[Export ("objectForKeyedSubscript:")]
+		NSObject ObjectForKeyedSubscript (string key);
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHMediaLibrary
+	{
+		[Export ("defaultLibrary", ArgumentSemantic.Strong)]
+		SHMediaLibrary DefaultLibrary { get; }
+
+		[Async]
+		[Export ("addMediaItems:completionHandler:")]
+		void Add (SHMediaItem[] mediaItems, Action<NSError> completionHandler);
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	interface SHSession
+	{
+		[Export ("catalog", ArgumentSemantic.Strong)]
+		SHCatalog Catalog { get; }
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		ISHSessionDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; set; }
+
+		[Export ("initWithCatalog:")]
+		IntPtr Constructor (SHCatalog catalog);
+
+		[Export ("matchStreamingBuffer:atTime:")]
+		void MatchStreamingBuffer (AVAudioPcmBuffer buffer, [NullAllowed] AVAudioTime time);
+
+		[Export ("matchSignature:")]
+		void Match (SHSignature signature);
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
```

Comment:
same, `[DisableDefaultCtor]`
the `.ctor` does not allow `null` `dataRepresentation`
and the `DataRepresentation` property does not allow null, nor allow you to set your own value